e2e/src/bin/historic_trades.rs

```diff
@@ -0,0 +1,276 @@
+    rename_all = "kebab"
+)]
+struct Options {
+    /// The events registry file store containing past exchange events.
+    #[structopt(long, env = "ORDERBOOK_FILE", parse(from_os_str))]
+    orderbook_file: PathBuf,
+
+    /// The output directory for the computed results.
+    #[structopt(long, env = "OUTPUT_DIR", default_value = "target", parse(from_os_str))]
+    output_dir: PathBuf,
+}
+
+fn main() -> Result<()> {
+    let options = Options::from_args();
+
+    let mut report = Report::new(File::create(options.output_dir.join("trades.csv"))?);
+    report.header()?;
+
+    cmd::for_each_batch(&options.orderbook_file, |history, batch| {
+        let auction_elements = history.auction_elements_for_batch(batch)?;
+        let settlement = history.settlement_for_batch(batch);
+
+        let new_orders = auction_elements
+            .iter()
+            .filter(|order| batch == order.valid.from)
+            .filter(|order| order.price.numerator != 0 && order.price.denominator > MIN_AMOUNT)
+            .filter(|order| order.balance > U256::from(MIN_AMOUNT));
+        for order in new_orders {
+            let settlement = settlement.as_ref();
+            let pricegraph = Pricegraph::new(
+                auction_elements
+                    .iter()
+                    .filter(|element| (element.user, element.id) != (order.user, order.id))
+                    .cloned(),
+            );
+
+            let meta = OrderMetadata::compute(settlement, order, &pricegraph);
+            let result = if meta.is_resonable_order() {
+                process_reasonable_order(settlement, order, &meta)
+            } else {
+                process_unreasonable_order(&meta)
+            };
+
+            report.record_result(batch, settlement, order, &meta, result)?;
+        }
+
+        Ok(())
+    })?;
+
+    report.summary()
+}
+
+struct OrderMetadata {
+    effective_sell_amount: f64,
+    limit_price: f64,
+    estimated_limit_price: Option<f64>,
+    settled_xrate: Option<f64>,
+    trade: Option<Trade>,
+}
+
+impl OrderMetadata {
+    fn compute(settlement: Option<&Settlement>, order: &Element, pricegraph: &Pricegraph) -> Self {
+        let effective_sell_amount =
+            pricegraph::num::u256_to_f64(order.balance).min(order.remaining_sell_amount as _);
+        let limit_price = order.price.numerator as f64 / order.price.denominator as f64;
+        let estimated_limit_price =
+            pricegraph.estimate_limit_price(order.pair, effective_sell_amount);
+        let settled_xrate =
+            settlement.and_then(|settlement| find_settled_exchange_rate(settlement, order.pair));
+        let trade = settlement.and_then(|settlement| find_trade(settlement, order));
+
+        OrderMetadata {
+            effective_sell_amount,
+            limit_price,
+            estimated_limit_price,
+            settled_xrate,
+            trade,
+        }
+    }
+
+    fn is_resonable_order(&self) -> bool {
```

Comment:
Add doc comment explaining what "reasonable" means here. Looking at the other code I think this refers to that the order should reasonably be matched based on price.